File: tienda_back/src/main/java/ar/store/italiana/tienda_back/tienda_back/models/Product.java
package ar.store.italiana.tienda_back.tienda_back.models;

import lombok.Data;

import javax.persistence.*;

@Data
@MappedSuperclass
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
public abstract class Product {
	protected Boolean enabled;
	protected String subcategory;
	protected String sex;
	protected String brand;
	protected String model;
	protected String code;
	protected Boolean stock;
	protected Float price;
	@Column(length = 350000)
	protected String image;
	protected Boolean isOnFavs;
}

File: tienda_front/src/app/models/product.ts
export interface Product{
    enabled:boolean,
    subcategory:string|null,
    sex:string,
    brand:string,
    model:string,
    code:string,
    stock:boolean,
    price:number,
    image:string,
    isOnFavs:boolean,
    category?:string;
}